pallet_error_naming_convention.md

```diff
@@ -0,0 +1,11 @@
+# Pallet Error Naming Conventions
+
+1) Use capitalized camel case in the variant names. For example, instead of using "NOT_FOUND" you should use "NotFound".
+
+2) Avoid using the word "error" as a suffix for the variant names. For example, instead of using "NotFoundError" you should use "NotFound". It's clear from the caller's context that this is an error.
+
+3) Avoid using the pallet name in the variant names. For example instead of "Web3NameNotFound" you should use "NotFound". 
+
+4) Try to take words from the vocabulary already defined in the code base. For example instead of introducing a new variant "NotExisting" you should use, once again, "NotFound"
```

Comment:
could you list the vocabulary here?
e.g. NotAuthorized, NotFound, AlreadyExists, MaxXYZExceeded (thats all? i guess...)